source/fundamentals/connection/connection-options.txt

```diff
@@ -0,0 +1,414 @@
+.. _csharp-connection-options:
+
+==================
+Connection Options
+==================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+This section describes the MongoDB connection and authentication options
+available in the {+driver-short+}. You can configure your connection using either 
+the connection URI or a ``MongoClientSettings`` object.
+
+.. _csharp-connection-uri:
+
+----------------------------------
+Using the Connection URI
+----------------------------------
+
+If you pass a connection URI to the ``MongoClient`` constructor, you can include
+connection options in the string as ``<name>=<value>`` pairs. In the following example,
+the connection URI contains the ``connectTimeoutMS`` option with a value of ``60000``
+and the ``tls`` option with a value of ``true``: 
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/LocalConnectionConfig.cs
+   :language: csharp
+   :dedent:
+
+.. _csharp-mongo-client-settings:
+
+-----------------------------
+Using ``MongoClientSettings`` 
+-----------------------------
+
+You can use a ``MongoClientSettings`` object to configure connection settings in code
+rather than in a connection URI. Configuring the connection this way makes it easier to 
+change settings at runtime, helps you catch errors during compilation, and provides 
+more configuration options than the connection URI.
+
+To use a ``MongoClientSettings`` object, create an instance of the class, set 
+its properties, and pass it as an argument to the ``MongoClient`` constructor:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/MongoClientSettingsConfig.cs
+   :language: csharp
+   :dedent:
+
+Connection Options
+------------------
+
+The following table lists each connection option available in the
+``MongoClientSettings`` class and, if possible, how to achieve the same result in 
+the connection string. If a ``MongoClientSettings`` property maps to more than one 
+option in the connection string, the **Connection URI Example** column shows all 
+relevant options. 
+
+.. note::
+
+  If you're using a query parameter for a time duration, the value must be in 
+  milliseconds. For example, to specify 60 seconds, use the value ``60000``. If you're 
+  using a ``MongoClientSettings`` object for a time duration, use the appropriate 
+  ``TimeSpan`` value.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 30 30
+
+   * - **MongoClientSettings** Property
+     - Connection URI Example
+     - Description
+
+   * - **AllowInsecureTls**
+     - ``tlsInsecure=true``
```

Comment:
should we mention restrictions on some options combinations like here https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L1139 (and in nearest lines)?

Reply:
I've added some. Let me know if you think it's enough.